tests/testthat/test-ard_categorical.R

```diff
@@ -868,3 +868,30 @@ test_that("ard_categorical() with hms times", {
     )$stat
   )
 })
+
+test_that("ard_categorical() errors with incomplete factor columns", {
+
+  # First check output is fine when there is a valid factor variable
+  expect_snapshot(
+    mtcars |>
+      dplyr::mutate(am = factor(am)) |>
+      ard_categorical(variables = am)
+  )
```

Comment:
For all the test chunks you've added, i think we can skip this first test because we already have tests to ensure vanilla factors are working properly.

Reply:
Removed the first test, just checking that vanilla factors work, from all of the added tests.